### PR TITLE
Added docs for sink topic partitions and replica defaults.

### PR DESCRIPTION
**Pull request by hjafarpour**
Docs update for the default partition and replica count for sink topics in queries.

### CODE REVIEW

File: docs/developer-guide/syntax-reference.rst [hunk 1]
```diff
@@ -446,12 +446,18 @@ The WITH clause for the result supports the following properties:
 |                         | set, then the format of the input stream/table is used.                                              |
 +-------------------------+------------------------------------------------------------------------------------------------------+
 | PARTITIONS              | The number of partitions in the backing topic. If this property is not set, then the number          |
-|                         | of partitions is taken from the value of the ``ksql.sink.partitions`` property, which                |
-|                         | defaults to four partitions. The ``ksql.sink.partitions`` property can be set in the                 |
-|                         | properties file the KSQL server is started with, or by using the ``SET`` statement.                  |
+|                         | of partitions of the input stream/table will be used. In the Join queries the properties would       |
```
Comment (JimGalasyn) [inline docs/developer-guide/syntax-reference.rst]: "In join queries, the property values are taken from the left-side stream or table."

I'd make the edits directly, but that would mess up the table alignment. :)

File: docs/installation/upgrading.rst [hunk 2]
```diff
@@ -6,6 +6,16 @@ Upgrading KSQL
...
 capacity to take over temporarily for unavailable, restarting servers.
 
+Upgrading from KSQL 5.2 to KSQL 5.3
+-----------------------------------
+
+Notable changes in 5.3:
+
+* Configuration:
+
+    * ksql.sink.partitions and ksql.sink.replicas are deprecated. All new queries will use the source topic partition count and replica count for the sink topic instead unless partitions and replicas are set in the WITH clause.
```
Comment (JimGalasyn) [inline docs/installation/upgrading.rst]:
```suggestion
    * ``ksql.sink.partitions`` and ``ksql.sink.replicas`` are deprecated. All new queries use the source topic partition count and replica count for the sink topic, unless you set partitions and replicas explicitly in the WITH clause.
```

File: docs/developer-guide/syntax-reference.rst [hunk 3]
```diff
@@ -446,12 +446,18 @@ The WITH clause for the result supports the following properties:
...
+|                         | of partitions of the input stream/table will be used. In join queries, the property values are taken |
+|                         | from the left-side stream or table.                                                                  |
+|                         | For KSQL 5.2 and earlier, if the property is not set, the value of the ``ksql.sink.partitions``      |
+|                         | property, which defaults to four partitions, will be used. The ``ksql.sink.partitions`` property can |
+|                         | be set in the properties file the KSQL server is started with, or by using the ``SET`` statement.    |
 +-------------------------+------------------------------------------------------------------------------------------------------+
 | REPLICAS                | The replication factor for the topic. If this property is not set, then the number of                |
-|                         | replicas of the input stream or table will be used.                                                  |
+|                         | replicas of the input stream or table will be used. In join queries, the property values are take    |
```
Comment (vcrfxia) [inline docs/developer-guide/syntax-reference.rst]:
```suggestion
|                         | replicas of the input stream or table will be used. In join queries, the property values are taken    |
```
nit: typo.

File: docs/installation/server-config/config-reference.rst [hunk 4]
```diff
@@ -160,18 +160,20 @@ will be prefixed as ``_confluent-ksql-default_`` (e.g. ``_command_topic`` become
 .. _ksql-sink-partitions:
 
 --------------------
-ksql.sink.partitions
+ksql.sink.partitions (Depricated)
```
Comment (vcrfxia) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
ksql.sink.partitions (Deprecated)
```
nit: spelling. Also three more occurrences in this file.

File: docs/installation/server-config/config-reference.rst [hunk 5]
```diff
@@ -160,18 +160,20 @@ will be prefixed as ``_confluent-ksql-default_`` (e.g. ``_command_topic`` become
...
 --------------------
 
 The default number of partitions for the topics created by KSQL. The default is four.
+This property has been depricated since 5.3 release. For more info see :ref:`KSQL Syntax Reference <ksql_syntax_reference>`.
```
Comment (vcrfxia) [inline docs/installation/server-config/config-reference.rst]: Can we provide a more specific link than the general syntax reference link? Ideally we'd link directly to the "partitions" row in the relevant table. If that's not possible, we can link to the table itself and call out that the user should be looking at the "partitions" row.

Same comment for "replicas" below.

Comment (hjafarpour) [reply docs/installation/server-config/config-reference.rst]: The table rows could not be referenced. I referenced the section and mentioned the WITH clause table instead.

**Pull request merged**